import "dos.proto";
package dos;

option cc_generic_services = true;
option py_generic_services = true;

message PollAgentRequest {}

message PollAgentResponse {
  optional NodeStatus status = 1;  
}

message RunPodRequest {
  optional string pod_name = 1;
  optional PodSpec pod = 3;
}

message RunPodResponse {
  optional RpcStatus status = 1;  
}

message DeletePodRequest {
  optional string name = 1;
  optional RpcStatus status = 2;
}

service Agent {
  rpc Poll(PollAgentRequest) returns (PollAgentResponse);
  rpc Run(RunPodRequest) returns (RunPodResponse);
  rpc Delete(DeletePodRequest) returns (DeletePodResponse);
}
